cdap-data-fabric/src/main/java/co/cask/cdap/data2/metrics/HBaseDatasetMetricsReporter.java

```diff
@@ -94,25 +102,52 @@ protected final ScheduledExecutorService executor() {
     return executor;
   }
 
-  private void reportHBaseStats() throws IOException {
-    Map<String, HBaseTableUtil.TableStats> tableStats = hBaseTableUtil.getTableStats(hAdmin);
+  private void reportHBaseStats() throws IOException, DatasetManagementException {
+    Map<TableId, HBaseTableUtil.TableStats> tableStats = hBaseTableUtil.getTableStats(conf, hAdmin);
     if (tableStats.size() > 0) {
       report(tableStats);
     }
   }
 
-  private void report(Map<String, HBaseTableUtil.TableStats> datasetStat) {
-    // we use "0" as runId: it is required by metrics system to provide something at this point
-    for (Map.Entry<String, HBaseTableUtil.TableStats> statEntry : datasetStat.entrySet()) {
-      Id.DatasetInstance datasetInstance = userDsNamespace.fromNamespaced(statEntry.getKey());
-      if (datasetInstance == null) {
-        // not a user dataset
+  private void report(Map<TableId, HBaseTableUtil.TableStats> tableStats) {
+    for (Map.Entry<TableId, HBaseTableUtil.TableStats> statEntry : tableStats.entrySet()) {
+      String namespace = statEntry.getKey().getNamespace().getId();
+      // emit metrics for only user datasets, namespaces in system and
+      // tableNames that doesn't start with user are ignored
+      if (namespace.equals("system") || !(statEntry.getKey().getTableName().startsWith("user"))) {
         continue;
       }
-      MetricsCollector collector =
-        metricsService.getCollector(ImmutableMap.of(Constants.Metrics.Tag.DATASET, datasetInstance.getId()));
-      // legacy format: dataset name is in the tag. See DatasetInstantiator for more details
-      collector.increment("dataset.size.mb", statEntry.getValue().getTotalSizeMB());
+      // for user tables, strip 'user.' prefix from tableName
+      String tableName = statEntry.getKey().getTableName();
+      tableName = tableName.substring(tableName.indexOf(".") + 1);
+
+      try {
+        Collection<DatasetSpecificationSummary> instances = dsFramework.getInstances(statEntry.getKey().getNamespace());
+        for (DatasetSpecificationSummary spec : instances) {
+          // todo :  we are stripping cdap.{namespace} right now , this can be removed after namespace fixes
+          String dsName = stripRootPrefixAndNamespace(spec.getName());
+          if (tableName.startsWith(dsName)) {
+            MetricsCollector collector =
+              metricsService.getCollector(ImmutableMap.of(Constants.Metrics.Tag.NAMESPACE, namespace,
+                                                          Constants.Metrics.Tag.DATASET, dsName));
+            collector.gauge("dataset.size.mb", statEntry.getValue().getTotalSizeMB());
+            break;
+          }
+        }
+      } catch (DatasetManagementException e) {
+        // No op
+      }
+    }
+  }
+
+  private String stripRootPrefixAndNamespace(String dsName) {
+    // ignoring "cdap." and namespace at beginning
+    String rootPrefix = conf.get(Constants.Dataset.TABLE_PREFIX) + ".";
```

Comment:
this is dataset name, i believe even in 2.8 we have cdap.namespace.datsetName , correct me if i am wrong.